viper.py

```diff
@@ -512,10 +516,25 @@ def fit_chunk(o, chunk, obsname, targ=None, tpltarg=None):
 
             if tplname:
                 p, e_p = S_mod.fit(x_ok, f_ok, v, a, bg, s,t, c=cc, c0=c0, dx=0.1, sig=sig[i_ok])
+            elif createtpl:
+                p, e_p = S_mod.fit(x_ok, f_ok, v, a, bg, s,t, c=cc, c0=c0, dx=0.1, sig=sig[i_ok], res=False, notell= True)
             else:
             # do not fit for velocity
                 p, e_p = S_mod.fit(x_ok, f_ok, a=a, b=bg, s=s,t=t0, c=cc, v0=0, c0=c0, dx=0.1, sig=sig[i_ok])
 
+    if createtpl:
+        S_tell = S_mod(x, *p)
+        S_tell /= np.nanmean(S_tell)
+
+        tf = f/S_tell
+        tf[S_tell<0.2] = 1
+        w1 = np.poly1d(p[2][::-1])(x-icen)
+
+        tpl_all[o,0][n] = w1 
+        tpl_all[o,1][n] = np.interp(w1 / (1+berv/c), w1, tf / np.nanmedian(tf))
```

Comment:
Is think interpolation is not needed here. `w1` is not a common grid for all observations.
How about something like:
```python
  tpl_all[o,0][n] = w1 * (1+berv/c)
  tpl_all[o,1][n] = tf / np.nanmedian(tf)
```

Reply:
Your right! As it gives the same results and saves computing time, I prefer it this way.

Reply:
Did some final tests again to make sure everything is working fine. While doing so I realized your option doesn't work, as in this way the barycentric correction is not applied in the right way and therefore the lines are shifted towards each other. Decided to keep it my way!

Reply:
I made my comment, before realising the crucials lines in
https://github.com/mzechmeister/viper/blob/d7a72df5bec7d89e36f3072f145a01795283661f/viper.py#L767-L768
For that you need the resampling.
We will go on with the resampling in this pull request.
But I will urge to refactor that and I open #26 to continue the discussion there.